utils/fixed_decimal/src/decimal.rs

```diff
@@ -552,6 +552,321 @@ impl FixedDecimal {
+    /// dec.truncate_right(-1);
+    /// assert_eq!("1234.5", dec.to_string());
+    /// ```
+    pub fn truncate_right(&mut self, n: i16) {
+        let bottom_magnitude = self.nonzero_magnitude_right();
+
+        if n <= self.lower_magnitude {
+            return;
+        } else if n <= bottom_magnitude {
+            self.lower_magnitude = {
+                if n <= 0 {
+                    n
+                } else {
+                    0
+                }
+            };
+        } else if n <= self.magnitude {
+            self.digits
+                .truncate((crate::ops::i16_abs_sub(self.magnitude, n).add(1)) as usize);
+            self.remove_trailing_zeros_from_digits_list();
+            self.lower_magnitude = {
+                if n <= 0 {
+                    n
+                } else {
+                    0
+                }
+            };
+        } else if n <= self.upper_magnitude {
+            self.lower_magnitude = {
+                if n <= 0 {
+                    n
+                } else {
+                    0
+                }
+            };
+            self.digits.clear();
+            self.magnitude = 0;
+        } else {
+            self.digits.clear();
+            self.lower_magnitude = 0;
+            self.magnitude = 0;
+            self.upper_magnitude = 0;
+        }
+
+        #[cfg(debug_assertions)]
+        self.check_invariants();
+    }
+
+    /// Ceils the number to the power ten of n.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use fixed_decimal::FixedDecimal;
+    /// # use std::str::FromStr;
+    ///
+    /// let mut dec = FixedDecimal::from_str("3.234").unwrap();
+    /// dec.ceil(0);
+    /// assert_eq!("4", dec.to_string());
+    ///
+    /// let mut dec = FixedDecimal::from_str("2.222").unwrap();
+    /// dec.ceil(-1);
+    /// assert_eq!("2.3", dec.to_string());
+    ///
+    /// let mut dec = FixedDecimal::from_str("22.222").unwrap();
+    /// dec.ceil(-2);
+    /// assert_eq!("22.23", dec.to_string());
+    ///
+    /// let mut dec = FixedDecimal::from_str("99.999").unwrap();
+    /// dec.ceil(-2);
+    /// assert_eq!("100.00", dec.to_string());
+    ///
+    /// let mut dec = FixedDecimal::from_str("99.999").unwrap();
+    /// dec.ceil(-10);
+    /// assert_eq!("99.9990000000", dec.to_string());
+    ///
+    /// let mut dec = FixedDecimal::from_str("-99.999").unwrap();
+    /// dec.ceil(-10);
+    /// assert_eq!("-99.9990000000", dec.to_string());
+    ///
+    /// let mut dec = FixedDecimal::from_str("-99.999").unwrap();
+    /// dec.ceil(-2);
+    /// assert_eq!("-99.99", dec.to_string());
+    ///
+    /// let mut dec = FixedDecimal::from_str("99.999").unwrap();
+    /// dec.ceil(10);
+    /// assert_eq!("10000000000", dec.to_string());
+    ///
+    /// let mut dec = FixedDecimal::from_str("-99.999").unwrap();
+    /// dec.ceil(10);
+    /// assert_eq!("-00000000000", dec.to_string());
+    /// ```
+    pub fn ceil(&mut self, n: i16) {
```

Comment:
Suggestion: internally you should consider having just 5 functions for "unsigned rounding mode" or "absolute rounding mode", and map the 9 signed rounding modes to those 5 functions according to the following table:

https://tc39.es/proposal-intl-numberformat-v3/out/numberformat/proposed.html#table-intl-unsigned-rounding-modes

Reply:
Great idea, let me do this in the next PR. lets get Ceil and Floor now and I will do that in the next PR.